src/lune/builtins/datetime/mod.rs

```diff
@@ -0,0 +1,222 @@
+use chrono::Month;
+use mlua::prelude::*;
+
+pub(crate) mod builder;
+pub(crate) mod date_time;
+
+use self::{
+    builder::DateTimeBuilder,
+    date_time::{DateTime, TimestampType, Timezone},
+};
+use crate::lune::util::TableBuilder;
+
+// TODO: Proper error handling and stuff
+
+pub fn create(lua: &'static Lua) -> LuaResult<LuaTable> {
+    TableBuilder::new(lua)?
+        .with_function("now", |_, ()| Ok(DateTime::now()))?
+        .with_function("fromUnixTimestamp", |lua, timestamp: LuaValue| {
+            let timestamp_cloned = timestamp.clone();
+            let timestamp_kind = TimestampType::from_lua(timestamp, lua)?;
+            let timestamp = match timestamp_kind {
+                TimestampType::Seconds => timestamp_cloned.as_i64().ok_or(LuaError::external("invalid float integer timestamp supplied"))?,
+                TimestampType::Millis => {
+                    let timestamp = timestamp_cloned.as_f64().ok_or(LuaError::external("invalid float timestamp with millis component supplied"))?;
+
+                    ((((timestamp - timestamp.fract()) as u64) * 1000_u64) // converting the whole seconds part to millis
+                    // the ..3 gets a &str of the first 3 chars of the digits after the decimals, ignoring
+                    // additional floating point accuracy digits
+                        + (timestamp.fract() * (10_u64.pow(timestamp.fract().to_string().split('.').collect::<Vec<&str>>()[1][..3].len() as u32)) as f64) as u64) as i64
+                    // adding the millis to the fract as a whole number
+                    // HACK: 10 * (timestamp.fract().to_string().len() - 2) gives us the number of digits
+                    // after the decimal
+                }
+            };
+
+            Ok(DateTime::from_unix_timestamp(timestamp_kind, timestamp))
+        })?
+        .with_function("fromUniversalTime", |lua, date_time: LuaValue| {
+            Ok(DateTime::from_universal_time(DateTimeBuilder::from_lua(date_time, lua).ok()))
+        })?
+        .with_function("toUniversalTime", |_, this: DateTime| {
+            Ok(this.to_universal_time())
+        })?
```

Comment:
No need for these on the exports table:

```lua
local DateTime = require("@lune/datetime")

-- this should not be valid
local dt = DateTime.now()
DateTime.toUniversalTime(dt)

-- this should be
local dt = DateTime.now()
dt:toUniversalTime()
```

If you remove them from the exports table the methods on the userdata itself will still be available

```suggestion
```